packages/flutter/lib/src/painting/inline_span.dart

```diff
@@ -364,6 +371,26 @@ abstract class InlineSpan extends DiagnosticableTree {
   @protected
   int? codeUnitAtVisitor(int index, Accumulator offset);
 
+  /// Returns an [InlineSpan] by applying `newAttributes` to a [TextRange] within
+  /// this span.
+  ///
+  /// The `textRange` argument is the range, in relation to the start of the root
+  /// span, to which the `newAttributes` should be applied.
+  ///
+  /// The `offset` argument is a mutable offset. When this implementation is
+  /// invoked, its value is set to the offset from the start of the root span to
+  /// the span of this [InlineSpan]. The implementation is responsible for
+  /// advancing the offset by this [InlineSpan]'s length in UTF16 code units,
+  /// such that the `offset` points to the end (enclusive) of this [InlineSpan]
+  /// when this method returns.
+  @protected
+  InlineSpan updateAttributesAtOffset(covariant InlineSpanAttributes newAttributes, TextRange textRange, Accumulator offset);
+
+  /// Creates a new [InlineSpan] from this [InlineSpan] by applying
+  /// `newAttributes` to the given [TextRange] within this span.
```

Comment:
Same here, maybe leave a comment that subclasses should override updateAttributesAtOffset instead of this one?